src/plugins/github/__snapshots__/graphql.test.js.snap

```diff
@@ -0,0 +1,494 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`graphql #postQueryExhaustive resolves a representative query 1`] = `
```

Comment:
so, this test case is very powerful, in that it tests the query construction, the fragment resolution, etc.
however, it also inlines the entire real query in the snapshot, which means that any time we modify the real query, we will generate a snapshot failure here. since we don't actually use the query, this is just noise/false positives for our snapshot detection.

is it feasible to re-write the test without causing spurious failures whenever we tweak the query? perhaps by constructing a small example query to use instead?

Reply:
Note: if we change the query, there is a fairly high probability we also need to update other snapshots, so having spurious failures on that change is not as bad as in the general case

Reply:
> however, it also inlines the entire real query in the snapshot, which means that any time we modify the real query, we will generate a snapshot failure here.

Correct.

> is it feasible to re-write the test without causing spurious failures whenever we tweak the query? perhaps by constructing a small example query to use instead?

I don’t think that constructing a small query is the right way to go. The pagination API is only intended to work on the standard query, so we would be violating the contract in the test—in an important way.

Here are two alternative options:
  - Remove the snapshot assertion entirely. This does not significantly affect the predictive power of the test: the `result` assertion is most important, and the `toHaveBeenCalledTimes(3)` makes sure that we aren’t making spurious requests.
  - Remove the first element from the snapshot array, and instead separately assert that the first argument was `{body: createQuery(), variables}`. This retains the full predictive power of the test, and shrinks the snapshot (but makes the code less uniform).
The reason that I included the snapshots was primarily for readers’ benefit, to make it easy to see what the continuation queries looked like. But, in retrospect, perhaps that’s not necessary: a reader who cares about this probably has a working copy of the code and enough understanding of the API to easily add a `console.log` to the right spot in the test. So I am fine with either of these solutions or the status quo, leaning toward removing the snapshot entirely. Thoughts?

Reply:
I think removing that snapshot assertion is fine. Making it convenient to read what the generated query looks like is nice, but not worth the cause of introducing spurious snapshot failures - to do so privileges an edge case (wanting to see the exact generated query) over the common case (modifying something and not wanting spurious snapshot failures).